src/settingsview.cpp

```diff
@@ -176,27 +180,23 @@ void SettingsView::setReopenTab(bool reopen)
 
 void SettingsView::onDownloadDirChanged(const QString &dir)
 {
-    ui->downloadDirPath->setText(formatDownloadDir(dir));
+    ui->downloadDirPath->setText(formatSettingsDir(dir));
     ui->downloadDirPath->setToolTip(dir);
 }
 
-void SettingsView::copyDownloadPathToClipboard()
+void SettingsView::copySettingsPathToClipboard(QPushButton* button, QString pathToCopy)
```

Comment:
1. The order of parameters is somewhat unnatural. `pathToCopy` is the primary parameter for the functionality of this procedure. `button` is passed so that it can also perform the nice-to-have (tooltip) feature.
2. Also please pass the button by `const` pointer.

Reply:
> The order of parameters is somewhat unnatural. pathToCopy is the primary parameter for the functionality of this procedure. button is passed so that it can also perform the nice-to-have (tooltip) feature.

Yeah I did that initially, do not why i changed it back...

> Also please pass the button by const pointer.

The `QToolTip::showText` function does not accept a const.

Reply:
@veloman-yunkan is it ok to keep as non const?

Reply:
> @veloman-yunkan is it ok to keep as non const?

Yes